hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestOnlineSlowLogProvider.java

```diff
@@ -0,0 +1,307 @@
+
+    int i = 0;
+
+    // add 5 records initially
+    for (; i < 5; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    List<SlowLogPayload> slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    Assert.assertEquals(slowLogPayloads.size(), 5);
+    for (int j = 0; j < 5; j++) {
+      confirmPayloadParams(j, j + 1, slowLogPayloads);
+    }
+
+    // add 2 more records
+    for (; i < 7; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+
+    // confirm ringbuffer has only one payload left
+    Assert.assertEquals(slowLogPayloads.size(), 7);
+    confirmPayloadParams(5, 6, slowLogPayloads);
+    confirmPayloadParams(6, 7, slowLogPayloads);
+
+    // add 3 more records
+    for (; i < 10; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    // confirm ringbuffer is full
+    Assert.assertEquals(slowLogPayloads.size(), 8);
+    confirmPayloadParams(7, 8, slowLogPayloads);
+    confirmPayloadParams(0, 9, slowLogPayloads);
+    confirmPayloadParams(1, 10, slowLogPayloads);
+
+    // add 4 more records
+    for (; i < 14; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    // confirm ringbuffer is full
+    Assert.assertEquals(slowLogPayloads.size(), 8);
+    confirmPayloadParams(2, 11, slowLogPayloads);
+    confirmPayloadParams(3, 12, slowLogPayloads);
+    confirmPayloadParams(4, 13, slowLogPayloads);
+
+    boolean isRingBufferCleaned = onlineSlowLogProvider.clearSlowLogPayloads();
+    Assert.assertTrue(isRingBufferCleaned);
+    LOG.debug("cleared the ringbuffer of Online Slow Log records");
+    slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    // confirm ringbuffer is empty
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithHighRecords() throws Exception {
+    Configuration conf = getOnlineSlowLogConf(14);
+    onlineSlowLogProvider = new OnlineSlowLogProvider(conf);
+    Assert.assertEquals(onlineSlowLogProvider.getSlowLogPayloads().size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    for (int i = 0; i < 4000; i++) {
```

Comment:
This seems like an arbitrary number of times to loop. How about reducing this number to something scaled off of the expected buffer capacity, say `* 3` or whatever multiple is necessary.